elasticsearch/json-adaptive-selection-6plus.yml

```diff
@@ -0,0 +1,65 @@
+type: json
+data:
+  url: /_nodes/_local/stats/indices,transport,http,thread_pool,jvm,process,fs,script,discovery,ingest,adaptive_selection?format=smile
+  server: ${ST_MONITOR_ES_NODE_HOSTPORT}
+  basicHttpAuthUsername: ${ST_MONITOR_ES_NODE_BASICAUTH_USERNAME}
+  basicHttpAuthPassword: ${ST_MONITOR_ES_NODE_BASICAUTH_PASSWORD}
+  smileFormat: true
+require:
+  - className: com.sematext.spm.client.es.EsVersionCheck
+    values: 6-*
+
+observation:
+  - name: adaptive_replica_selection
+    metricNamespace: es
+    path: $.nodes.${nodeId}.adaptive_selection.${remoteNodeId}
+
+    metric:
+      - name: adaptiveReplicaSelection.searches.outgoing
+        source: outgoing_searches
+        type: long_gauge
+        label: outgoing searches
+        description: outgoing searches
+
+      - name: adaptiveReplicaSelection.queue.size.avg
+        source: avg_queue_size
+        type: double_gauge
+        label: average queue size
+        description: average queue size
+
+      - name: adaptiveReplicaSelection.service.time.avg
+        source: avg_service_time_ns
+        type: long_gauge
+        label: average service time
+        description: average service time
+        unit: ns
+
+      - name: adaptiveReplicaSelection.response.time.avg
```

Comment:
Yes :( But I think it's more than nothing. It can be something that evolves. The thing is, we don't really care for the average here, average is just a shortcut. An EWMA (Exponentially Weighted Moving Average) is being computed. For example: https://github.com/elastic/elasticsearch/pull/24989

So we only really care how that value looks in a moment. That is one of the big factors in computing the rank of a node (i.e. how likely it is to be selected for the next query).

Reply:
But there is always aggregation on our backend on top of whatever value is reported by the agent. So user would end up with avg(avg(response_time)), while thinking he sees avg(response_time).

> So we only really care how that value looks in a moment.

Yes, we care about that, but that is not what user will see, there is no "in a moment" in SC.

Reply:
I think if we take the average over e.g. 5 minutes, it still gives the user a good enough idea. In the end, what matter is whether load/queries are imbalanced across nodes and why. If these values are close to each others, we're looking at a healthy cluster. If they are further apart, we may have a problem. So I think it's OK as it is. Not ideal, but OK :)

Reply:
> I think if we take the average over e.g. 5 minutes, it still gives the user a good enough idea. 

We can't "take" anything. It is what it is. We record its value every 10 seconds and backend/UI aggregate it on granularity currently used in chart.

Reply:
Right, right. By "take" I meant the aggregation on the granularity of 5 minutes, as an example.